### PR TITLE
Add github action for secure integ tests

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
Adding github action to run integ tests against cluster with security plugin. Approach is mainly taken from AD plugin https://github.com/opensearch-project/anomaly-detection/blob/main/.github/workflows/test_security.yml. Currently we're not able to build artifacts for native libraries in our regular assemble task, so for this iteration will be using latest pre-built versions from docker image. This means they may be different from PR version in case we're changing those libs in scope of PR.  We'll be running tests on linux as this gives sufficient confidence, and for macos/win the docker setup has some difficulties.

PR requires https://github.com/opensearch-project/k-NN/pull/827  and https://github.com/opensearch-project/k-NN/pull/847 to be merged

### Issues Resolved
https://github.com/opensearch-project/security/issues/2265
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/test_security.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+name: Security test workflow for k-NN
```
Comment (navneet1v) [inline .github/workflows/test_security.yml:1]: Rename this to, test k-NN on Secure Cluster

File: .github/workflows/test_security.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+on:
```
Comment (navneet1v) [inline .github/workflows/test_security.yml:2]: we should also make this as periodic github action.

File: .github/workflows/test_security.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    branches:
+      - "*"
+      - "feature/**"
+
+jobs:
+  Build-ad:
+    strategy:
+      matrix:
+        java: [ 11,17 ]
+        os: [ubuntu-latest]
```
Comment (navneet1v) [inline .github/workflows/test_security.yml:17]: should we add mac and windows?

File: .github/workflows/test_security.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      fail-fast: false
```
Comment (navneet1v) [inline .github/workflows/test_security.yml:18]: why this is false? I think we should make it true.

File: .github/workflows/test_security.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    name: Security test workflow for k-NN Plugin
```
Comment (navneet1v) [inline .github/workflows/test_security.yml:20]: update the name as suggested above.

File: .github/workflows/test_security.yml (hunk 3 above)
Comment (martin-gaievski) [reply .github/workflows/test_security.yml:17]: I'm not sure that will benefit us as running on at least one platform gives lot of confidence, also it's not easy due to docker setup on those platforms. We can add later if needed

File: .github/workflows/test_security.yml (hunk 5 above)
Comment (martin-gaievski) [reply .github/workflows/test_security.yml:20]: ack

File: .github/workflows/test_security.yml (hunk 2 above)
Comment (martin-gaievski) [reply .github/workflows/test_security.yml:2]: ack

File: .github/workflows/test_security.yml (hunk 1 above)
Comment (martin-gaievski) [reply .github/workflows/test_security.yml:1]: ack

File: .github/workflows/test_security.yml (hunk 4 above)
Comment (martin-gaievski) [reply .github/workflows/test_security.yml:18]: ack

**Review by navneet1v — APPROVED** [review]

**Review by jmazanec15 — DISMISSED** [review]

**Pull request merged**